### PR TITLE
Make Spec compile on win32

### PR DESCRIPTION
**Pull request by RX14**
This makes the spec runner work on win32, and also fixes a few issues in the windows File port caught by the addition of exception support.

### CODE REVIEW

File: src/spec/dsl.cr [hunk 1]
```diff
@@ -1,6 +1,5 @@
 require "colorize"
 require "option_parser"
-require "signal"
```
[the remark this answers is not in the document]
Comment (RX14) [reply src/spec/dsl.cr]: signal is in the prelude so this require is fine to go anyway, but yeah adding a TODO to the `flag?` is good

File: spec/std/file_spec.cr [hunk 2]
```diff
@@ -967,7 +979,7 @@ describe "File" do
       File.writable?("foo\0bar")
     end
 
-    it_raises_on_null_byte "executable?" do
+    it_raises_on_null_byte "executable?", pending_win32: true do
```
Comment (straight-shoota) [inline spec/std/file_spec.cr]: Just duplicating the content of `it_raises_on_null_byte` wrapped in `pending_win32 do ... end` would be much simpler code than adding an extra argument for a single use.

**Comment by RX14** [comment]
Sorry for the extended delay getting this PR up to date!

File: spec/std/file_spec.cr [hunk 3]
```diff
@@ -967,8 +974,10 @@ describe "File" do
       File.writable?("foo\0bar")
     end
 
-    it_raises_on_null_byte "executable?" do
-      File.executable?("foo\0bar")
+    pending "errors on executable?" do
```
Comment (straight-shoota) [inline spec/std/file_spec.cr]: pending_win32

Comment (RX14) [reply spec/std/file_spec.cr]: Shows me not to commit at 1am...

File: src/spec.cr [hunk 4]
```diff
@@ -118,6 +118,9 @@ if ENV["SPEC_VERBOSE"]? == "1"
   Spec.override_default_formatter(Spec::VerboseFormatter.new)
 end
 
-Signal::INT.trap { Spec.abort! }
+{% unless flag?(:win32) %}
+  # TODO(windows): re-enable this once Signal is ported
```
Comment (straight-shoota) [inline src/spec.cr]: I don't think there is any precedent for `TODO(windows)` notation. Is this really a good idea?

`TODO: (windows)`?  should be easier to parse.

Comment (RX14) [reply src/spec.cr]: `TODO(windows)` is easier to grep than `/TODO:?\s+(windows)/`

Comment (straight-shoota) [reply src/spec.cr]: AFAIK it is general consensus that the identifier is followed by either a colon or whitespace. Adding parenthesis in between won't match `/TODO[: ]/` or similar regular expressions which are pretty commonly used for this kind of thing.

Comment (RX14) [reply src/spec.cr]: just grep TODO, i've never seen anyone make the regex more complicated

Comment (straight-shoota) [reply src/spec.cr]: Take a look at tools like `rake notes` and similar. They expect a certain format, and often it includes the name followed by colon or whitespace.

I think it is a good idea to add context to comment annotations. But it shouldn't add additional complexity to the simple `# TODO: message` format. Just add the context tag as part of the message and everything is fine.

Comment (RX14) [reply src/spec.cr]: The format I used is common, I didn't invent it. I'm sick and tired of making nitpicky corrections, especially when they're hypotheticals about fucking todo comments.

Comment (straight-shoota) [reply src/spec.cr]: I've never encountered this format and specifically asked for precedents. You didn't mention anything about it being used elsewhere until now.

Comment (Sija) [reply src/spec.cr]: @straight-shoota I've seen `TODO(username): message` format before, see for instance docs for https://github.com/pgilad/leasot or https://github.com/asticode/go-astitodo. Usually what's put inside of brackets is a _username_ but that's just a convention. It's noticably less common than `TODO: message` but still.

**Comment by ysbaddaden** [comment]
Nothing I guess? Let's merge?

**Comment by RX14** [comment]
@ysbaddaden I was waiting for another review actually after modifying the PR

**Pull request merged**